sale_invoice_relation_line/account_invoice.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Chafique Delli
+#    Copyright 2014 Akretion SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+
+
+class AccountInvoiceLine(orm.Model):
+    _inherit = 'account.invoice.line'
+    _order = 'sale_parent_line_id, name desc'
+
+    def _get_invoice_line_from_order_line(self, cr, uid, ids, context=None):
+        invoice_line_ids = self.pool['account.invoice.line'].search(
+            cr, uid, [('sale_line_id', 'in', ids)], context=context)
+        return invoice_line_ids
+
+    _columns = {
+        'sale_line_id': fields.many2one(
```

Comment:
Relation between invoice lines and order lines already exists (many2many)
https://github.com/odoo/odoo/blob/7.0/addons/sale/sale.py#L685

If the purpose is to make the related SO lines available on invoice line model, I would suggest to add the inverse many2many, using the existing table ```sale_order_line_invoice_rel```